### PR TITLE
chore: Add Mirai framework and Melos

### PR DESCRIPTION
**Pull request by divyanshub024**
## Description

This PR extracts some common framework files to mirai_framework package. And add Melos to manage local packages.

## Type of Change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Code refactor
- [x] Build configuration change
- [ ] Documentation
- [x] Chore

### CODE REVIEW

File: examples/mirai_gallery/test/widget_test.dart
Comment (i-asimkhan) [inline examples/mirai_gallery/test/widget_test.dart:1]: Why are we deleting this file?

File: packages/mirai/pubspec.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ dependencies:
   json_annotation: ^4.8.1
   logger: ^1.4.0
   dio: ^5.2.0+1
+  mirai_framework: ^0.0.1
```
Comment (i-asimkhan) [inline packages/mirai/pubspec.yaml:18]: Can't we keep the localized version here as well? just like here https://github.com/Securrency-OSS/mirai/blob/9c453caca01b8dc66e502b1f84e0f65d59db7293/examples/mirai_gallery/pubspec.yaml

File: packages/mirai_webview/lib/parsers/mirai_webview/mirai_webview_parser.dart [hunk 2]
```diff
@@ -49,3 +49,16 @@ class _WebViewState extends State<_WebView> {
     return WebViewWidget(controller: _controller);
   }
 }
+
+extension ColorExt on String? {
```
Comment (i-asimkhan) [inline packages/mirai_webview/lib/parsers/mirai_webview/mirai_webview_parser.dart:53]: Can't we use now this from the **MIrai**

File: examples/mirai_gallery/test/widget_test.dart
Comment (divyanshub024) [reply examples/mirai_gallery/test/widget_test.dart:1]: These are not correct tests.

File: packages/mirai/pubspec.yaml (hunk 1 above)
Comment (divyanshub024) [reply packages/mirai/pubspec.yaml:18]: This is localized only. Melos overrides it.

File: packages/mirai_webview/lib/parsers/mirai_webview/mirai_webview_parser.dart (hunk 2 above)
Comment (divyanshub024) [reply packages/mirai_webview/lib/parsers/mirai_webview/mirai_webview_parser.dart:53]: Since Mirai is not used. We can't call it

**Review by i-asimkhan — APPROVED** [review]
LGTM!

**Pull request merged**